third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.c

```diff
@@ -615,13 +618,125 @@ void chpl_task_exit(void)
 #endif /* QTHREAD_MULTINODE */
 }
 
+//
+// Tasking callback support.
```

Comment:
Ah ok. 

I wonder if you could split up do_callbacks into 2 routines. One that builds the chpl_task_cb_info_t that is tasking layer specific and the other that actually calls each registered callback that tasks the completed chpl_task_cb_info_t.

Though that could be pretty annoying since you'd need to pass along if you need to build the full on id_only version, so maybe it is just simpler to have do_callbacks be implementation specific even if some of the code is identical..

Anyways, not something that needs to be changed for this PR.